lib/unifex/specs_dsl.ex

```diff
@@ -164,6 +164,27 @@ defmodule Unifex.Specs.DSL do
     store_config(:callback, {hook, fun})
   end
 
+  @doc """
+  Defines struct type specification
+
+  The specification should be in the form of
+
+      struct_def my_struct :: %My.Struct{
+        field1: type1,
+        field2: type2,
+        ...
+      }
+
+  Struct specified in such way can be used in like any other supported type, E.g.
+
+      spec my_function(in_struct :: my_struct) :: {:ok :: label, out_struct :: my_struct}
+
+  Elixir definition of %My.Struct{} should contain every field listed in specification and is not generated by Unifex
+  """
+  defmacro struct_def(struct) do
```

Comment:
According to #60 let's rename it to `struct`

Reply:
`struct` is name of already existing function in `Kernel` module

https://hexdocs.pm/elixir/Kernel.html#struct/1

Reply:
Ahh right, `def_struct` has been already taken too. I am not convinced about `struct_def` as it breaks convention (we probably always have `def_xxx`). Maybe `def_c_struct` or `def_unifex_struct`? @mat-hek do you have any ideas?

Reply:
Maybe we could just have `type` use it also for enums? It would reflect elixir `@type` like `spec` reflects `@spec`